src/miner.cpp

```diff
@@ -582,6 +582,24 @@ CBlockTemplate* CreateNewBlockWithKey(CReserveKey& reservekey, CWallet* pwallet,
     if (!reservekey.GetReservedKey(pubkey))
         return NULL;
 
+    const int nHeightNext = chainActive.Tip()->nHeight + 1;
+    static int nLastPOWBlock = Params().LAST_POW_BLOCK();
+
+    // If we're building a premature PoS block, don't continue
+    if ((nHeightNext < nLastPOWBlock) && fProofOfStake) {
```

Comment:
shouldn't this be `nHeightNext <= nLastPOWBlock` ?

Reply:
Very good question; and hitting me right in my gut of "what's right to do here".  First.  Yes; you're correct it should be.  I went with < because of the fluid situation with block transitions that's been being cleaned up in other PRs as well, and how some things are one block off.

I had considered completely removing the PoW phase check, since BitcoinMiner() should prevent it from coming down in this code anyway.  

I'm leaning back towards completely removing the block of code; BitcoinMiner() should cover it, and if it really slips by, it's not going to do anything different then the original code did... try to create a block that gets rejected.  Will think about it today and address one way or the other tonight (`<=` vs. no fProofOfStake check)

Reply:
```
        if (fProofOfStake) {
            if (chainActive.Tip()->nHeight < Params().LAST_POW_BLOCK()) {
                //  The last PoW block hasn't even been mined yet.
                MilliSleep(Params().TargetSpacing() * 1000);       // sleep a block
                continue;
            }
```
The above code in bitcoinMiner() will prevent any premature PoS block creation from happening; therefore the check in CreateNewBlockWithKey() for the same thing is redundant and will not be able to be hit.  I removed it.